pkg/filter/requestadaptor/requestadaptor.go

```diff
@@ -81,6 +88,15 @@ func (ra *RequestAdaptor) Results() []string {
 // Init initializes RequestAdaptor.
 func (ra *RequestAdaptor) Init(filterSpec *httppipeline.FilterSpec) {
 	ra.filterSpec, ra.spec = filterSpec, filterSpec.FilterSpec().(*Spec)
+	if ra.spec.Decompress != "" && ra.spec.Decompress != "gzip" {
+		panic("RequestAdaptor only support decompress type of gzip")
+	}
+	if ra.spec.Compress != "" && ra.spec.Compress != "gzip" {
+		panic("RequestAdaptor only support decompress type of gzip")
+	}
+	if ra.spec.Compress != "" && ra.spec.Decompress != "" {
+		panic("RequestAdaptor can only do compress or decompress for given request body, not both")
+	}
```

Comment:
```suggestion
	if ra.spec.Compress != "" && ra.spec.Decompress != "" {
		panic("RequestAdaptor can only do compress or decompress for given request body, not both")
	}
	if ra.spec.Body != "" && ra.spec.Decompress != "" {
		panic("No need to decompress when body is specified in RequestAdaptor spec")	
	}
```

Reply:
i didn't panic it because set `Body` and `Decompress` same time will not cause problem, but in this case `Decompress` will not run in this case. But your change is better to reduce confusion.